cypress-io/cypress
# chore: refactor app.yml config

- Closes #22994 

### User facing changelog

n/a

### Additional details

The current `config/app.yml` needed to be refactored due to containing many obsolete values and it wasn't included in the snapshot since yaml files do not get snapshotted.
* Converted `app.yml` to `app.json`
  * Removed obsolete fields:
    * `desktop_url`
    * `desktop_manifest_url`
    * `chromium_url`
    * `chromium_manifest_url`
    * `on_url`
  * Moved `cdn_url` field directly to `upload.js` since the value doesn't change between environments.
* Removed `konfig` module as it seemed like overkill for loading a single config value.

### Steps to test

* create a prod v8 snapshot (`DEBUG=cypress:snap*(info|debug|error) yarn build-v8-snapshot-prod`)
* Verify the `app.json` content in the snapshot
* Verify Cypress can still interact with the dashboard

### How has the user experience changed?

n/a

### PR Tasks


- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
